config/default/kustomization.yaml

```diff
@@ -27,67 +27,184 @@ resources:
 # [METRICS] Expose the controller manager metrics service.
 - manager_metrics_service.yaml
 
-patchesStrategicMerge:
+patches:
 
 # [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
 # crd/kustomization.yaml
-- manager_webhook_patch.yaml
+- path: manager_webhook_patch.yaml
 
 # Other configurations
-- manager_config_patch.yaml
+- path: manager_config_patch.yaml
 
-- manager_metrics_patch.yaml
+- path: manager_metrics_patch.yaml
 
 # [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER'.
 # Uncomment 'CERTMANAGER' sections in crd/kustomization.yaml to enable the CA injection in the admission webhooks.
 # 'CERTMANAGER' needs to be enabled to use ca injection
-#- webhookcainjection_patch.yaml
+#- path: webhookcainjection_patch.yaml
 
 #  Uncomment the patches line if you enable Metrics and CertManager
 # [METRICS WITH CERTMANGER] To enable metrics protected with certmanager, uncomment the following line.
 # This patch will protect the metrics with certmanager self-signed certs.
-#- certmanager_metrics_manager_patch.yaml
+#- path: certmanager_metrics_manager_patch.yaml
 
 # [PROMETHEUS WITH CERTMANAGER] to enable prometheus with certmanager, ../prometheus dir should be configured
 
 # the following config is for teaching kustomize how to do var substitution
-#vars:
-# [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER' prefix.
-#- name: CERTIFICATE_NAMESPACE # namespace of the certificate CR
-#  objref:
-#    kind: Certificate
-#    group: cert-manager.io
-#    version: v1
-#    name: serving-cert # this name should match the one in certificate.yaml
-#  fieldref:
-#    fieldpath: metadata.namespace
-#- name: CERTIFICATE_NAME
-#  objref:
-#    kind: Certificate
-#    group: cert-manager.io
-#    version: v1
-#    name: serving-cert # this name should match the one in certificate.yaml
-#- name: SERVICE_NAMESPACE # namespace of the service
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: webhook-service
-#  fieldref:
-#    fieldpath: metadata.namespace
-#- name: SERVICE_NAME
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: webhook-service
-#- name: METRICS_SERVICE_NAMESPACE # namespace of the service
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: controller-manager-metrics-service
-#  fieldref:
-#    fieldpath: metadata.namespace
-#- name: METRICS_SERVICE_NAME
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: controller-manager-metrics-service
+#replacements:
```

Comment:
Our CI does not test this flow.

Can you do a manual test to confirm this works?

Reply:
After uncommented the all cert-manager related configurations, I ran `kustomize build` and saw that the output is expected. If you still want me to test it on a live cluster, I can do it and let you know the results?.

Reply:
Yea, I think having an e2e test execute cert manager would be useful but we don't have that yet so I just want to make sure these configs will actually generate a functionally cert manager + jobset deployment.

Reply:
Cert-manager enabled e2e testing (i.e. pull-jobset-test-e2e-main-cert-manager) is on my TODO list in both JobSet and LWS. I think, we can execute the same e2e test with the only difference that cert-manager is enabled instead of internal cert management. But I don't know where the configuration locates in upstream?.

Reply:
I don't fully understand.

https://github.com/kubernetes-sigs/jobset/blob/main/config/components/manager/controller_manager_config.yaml#L7

I think you could toggle that to false and that should use CertManager for kustomize.

Reply:
Would it make sense if we have a new CI job that runs the same e2e tests but instead of using the internal cert management, it installs jobset with cert manager and all related patches?

Reply:
yea, that was something we did for Kueue.

Reply:
Thanks. I will test this pr tomorrow and let you know the results. Independently I am going to work on a pr for the new job.

Reply:
@kannon92 I've tested this on a live cluster. It works without any issue.